core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -813,6 +814,25 @@ default Optional<TrinoPrincipal> getSchemaOwner(ConnectorSession session, String
         return getSchemaOwner(session, new CatalogSchemaName("invalid", schemaName));
     }
 
+    /**
+     * Attempt to push down an update operation into the connector. If a connector
+     * can execute an update for the table handle on its own, it should return a
+     * table handle, which will be passed back to {@link #executeUpdate} during
+     * query executing to actually execute the update.
+     */
+    default Optional<ConnectorTableHandle> applyUpdate(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        return Optional.empty();
+    }
+
+    /**
+     * Execute the update operation on the handle returned from {@link #applyUpdate}.
+     */
+    default OptionalLong executeUpdate(ConnectorSession session, ConnectorTableHandle handle, Map<String, Domain> assignments)
```

Comment:
This is backwards. The `applyUpdate` operation should include all the parameters needed to represent the pushed-down update operation (e.g., the values that are being applied to that table as part of the update).

The `executeUpdate` operation should just take the TableHandle returned by `applyUpdate` and execute the corresponding update operation.

Reply:
Refactored : 
https://github.com/trinodb/trino/pull/16445/commits/1b630929edce27c1afb2cc1dc630aca5596981d1

Is it correct way?